spec/features/support/providers/users/adding_user_to_provider_spec.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+feature "Adding user to provider as an admin" do
+  let(:user) { create(:user, :admin) }
+
+  before do
+    given_i_am_authenticated(user:)
+    and_there_is_a_provider
+  end
+
+  describe "Adding user to organisation" do
+    scenario "With valid details" do
+      given_i_visit_the_support_provider_users_index_page
+      and_the_user_i_want_to_add_has_not_already_been_added
+      and_i_click_add_user
+      and_i_fill_in_first_name
+      and_i_fill_in_last_name
+      and_i_fill_in_email
+      and_i_continue
+      then_i_should_be_on_the_check_page
+      and_the_user_should_not_be_added_to_the_database
+
+      when_i_click_change_first_name
+      and_i_enter_a_new_first_name
+      and_i_continue
+      and_i_click_add_user
+
+      then_i_should_see_the_users_name_listed
+      and_i_should_see_the_users_email_listed
+      and_the_user_should_be_added_to_the_database
+    end
+
+    scenario "With invalid details" do
+      given_i_visit_the_support_provider_users_new_page
+      and_i_continue
+
+      then_i_should_see_the_error_summary
+    end
+  end
+
+  def and_there_is_a_provider
+    @provider = create(:provider, :with_users, provider_name: "School of bats")
+  end
+
+  def given_i_visit_the_support_provider_users_index_page
+    visit support_provider_users_path(provider_id: @provider.id)
+  end
+
+  def and_i_continue
+    support_users_new_page.submit.click
+  end
+
+  def given_i_visit_the_support_provider_users_new_page
+    support_users_new_page.load(provider_id: @provider.id)
+  end
+
+  def and_i_fill_in_first_name
+    support_users_new_page.first_name.set("Asa")
+  end
+
+  def and_i_fill_in_last_name
+    support_users_new_page.last_name.set("Bernhard")
+  end
+
+  def and_i_fill_in_email
+    support_users_new_page.email.set("viola_fisher@boyle.io")
+  end
+
+  def and_i_click_add_user
+    support_users_check_page.add_user.click
+  end
+
+  def then_i_should_be_on_the_check_page
+    expect(page).to have_current_path("/support/providers/#{@provider.id}/users/check")
```

Comment:
This (and the methods below) should probably be asserting against the check page object

Reply:
Done in [d3c9a061cdd9066c1ae30c17a6dc504d75ac6138](https://github.com/DFE-Digital/publish-teacher-training/pull/2839/commits/d3c9a061cdd9066c1ae30c17a6dc504d75ac6138)

Does this make the test faster or something or is it just a pattern we are following?

Reply:
It makes the test a little more clear as to what is being asserted against. You're communicating that the check page is what is expected here and asserting against it.